ads/google/a4a/test/test-google-ads-a4a-config.js

```diff
@@ -310,6 +312,261 @@ describe('a4a_config', () => {
+      element.setAttribute('type', 'doubleclick');
+      const preLaunch =
+        expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      win.pageExperimentBranches['expDoubleclickA4A'] = preLaunch.experiment;
+      let externalBranches, internalBranches;
+      if (isExperimentOn(win, 'a4aFastFetchDoubleclickLaunched')) {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      } else {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      }
+      expect(googleAdsIsA4AEnabled(win, element, 'expDoubleclickA4A',
+        externalBranches, internalBranches)).to.be.true;
+    });
+  it('should serve Fast Fetch to launched DoubleClick filler',
+    () => {
+      toggleExperiment(win, 'a4aFastFetchDoubleclickLaunched', true, true);
+      element.setAttribute('type', 'doubleclick');
+      let externalBranches, internalBranches;
+      if (isExperimentOn(win, 'a4aFastFetchDoubleclickLaunched')) {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      } else {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      }
+      expect(googleAdsIsA4AEnabled(win, element, 'expDoubleclickA4A',
+        externalBranches, internalBranches)).to.be.true;
+    });
+  it('should serve Fast Fetch to launched DoubleClick control',
+    () => {
+      toggleExperiment(win, 'a4aFastFetchDoubleclickLaunched', true, true);
+      toggleExperiment(win, 'expDoubleclickA4A', true, true);
+      element.setAttribute('type', 'doubleclick');
+      win.pageExperimentBranches['expDoubleclickA4A'] =
+        expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH.control;
+      let externalBranches, internalBranches;
+      if (isExperimentOn(win, 'a4aFastFetchDoubleclickLaunched')) {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      } else {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      }
+      expect(googleAdsIsA4AEnabled(win, element, 'expDoubleclickA4A',
+        externalBranches, internalBranches)).to.be.true;
+    });
+  it('should serve Delayed Fetch to launched DoubleClick experiment (holdback)',
```

Comment:
These are all very similar in structure.  Would it be practical to factor it out into a loop over test config vars?  Something like,

```javascript
[[DC_INTERNAL_PRE_LAUNCH, DC_INTERNAL_POST_LAUNCH, true, false, true],
 ...
].forEach(test_state => {
  const internal = test_state[0];
  const external = test_state[1];
  const doubleclick_launched = test_state[2];
  const adsense_launched = test_state[3];
  const expected = test_state[4];
  it(`should serve ${expected ? 'FF' : 'DF'} when doubleclick ${doubleclick_launched ? 'is' : 'is not'} launched`, () => {
    toggleExperiment(win, 'a4aFastFetchDoubleclickLaunched', doubleclick_launched, true);
    toggleExperiment(win, 'a4aFastFetchAdSenseLaunched', adsense_launched, true);
    expect(googleAdsIsA4AEnabled(win, element, 'expDoubleclickA4A', external, internal)).to.equal(expected);
  });
});
```